Products/ZenEvents/zentrap.py

```diff
@@ -762,71 +766,79 @@ def decode(value):
         """
 
         decoders = [Decoders.oid,
+                    Decoders.number,
                     Decoders.basestring,
                     Decoders.ipaddress,
                     Decoders.dateandtime,
                     Decoders.encode_base64]
 
-        for decoder in decoders:
-            out = decoder(value)
-            if out:
-                return out
+        try:
+            for decoder in decoders:
+                out = decoder(value)
+                if out or out == 0:
```

Comment:
To be more inclusive of possible valid values, this should probably be: `if out is not None:`
After all, we only _don't_ have a value when `out` is `None`.